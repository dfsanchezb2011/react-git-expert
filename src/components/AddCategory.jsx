import { useState } from "react"
import PropTypes from "prop-types"

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('');

    // {target} equivale a la desestructuración de event.target
    const onInputChange = ({ target }) => setInputValue(target.value);
    /* const onInputChange = (event) => setInputValue(event.target.value); */

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length < 1) {
            return;
        }
        /* onNewCategory(categories => [inputValue, ...categories]); */

        setInputValue('');
        onNewCategory(inputValue.trim());
    }

    return (
        <form onSubmit={onSubmit}>{/* <form onSubmit={(event) => onSubmit(event)}> */}
            <input
                type="text"
                placeholder="Buscar Gifs"
                value={inputValue}

                // Ambas líneas hacen lo mismo
                // Pero se reduce el código al saber que onChange por defecto manda el objeto event
                // value={inputValue} onChange={(event) => onInputChange(event)}
                onChange={onInputChange}

            />
        </form>
    )
};

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}