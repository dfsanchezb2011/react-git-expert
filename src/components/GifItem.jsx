export const GifItem = ({ title, urlImg, id }) => {

    return (
        <div className="card">
            <img src={urlImg} alt={title} />
            <p>{title}</p>
        </div>
    )
}