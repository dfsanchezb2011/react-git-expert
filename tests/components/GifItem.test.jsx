import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem";

describe('GifItem Component Tests', () => {

    let image = {
        title: 'Mash Up Pokemon GIF by Mic',
        urlImg: 'https://media0.giphy.com/media/Yrjxrzz4uFENG/giphy.gif?cid=dbe5ecd7ijba4oi6kobsj10vnl07gv2eoad4x0pad3j0i3ln&ep=v1_gifs_search&rid=giphy.gif&ct=g',
        /* id: 'Yrjxrzz4uFENG', */
    }

    test('Test 1. Snapshot - Must match with the snapshot', () => {
        const { container } = render(
            <GifItem {...image} />
        );

        expect(container).toMatchSnapshot();
    });

    test('Test 2. Must show the image with the props passed (URL and ALT)', () => {
        render(<GifItem {...image} />);
        /* screen.debug(); */
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(image.urlImg);
        expect(alt).toBe(image.title);
    });

    test('Test 3. Must show the title inside the component', () => {
        render(<GifItem {...image} />);
        expect(screen.getByText(image.title)).toBeTruthy();
    })
});