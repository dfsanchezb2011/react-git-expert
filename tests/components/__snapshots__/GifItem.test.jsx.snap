// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GifItem Component Tests Test 1. Snapshot - Must match with the snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Mash Up Pokemon GIF by Mic"
      src="https://media0.giphy.com/media/Yrjxrzz4uFENG/giphy.gif?cid=dbe5ecd7ijba4oi6kobsj10vnl07gv2eoad4x0pad3j0i3ln&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Mash Up Pokemon GIF by Mic
    </p>
  </div>
</div>
`;
